.github/workflows/build-publish.yml

```
@@ -0,0 +1,87 @@
+name: Build and publish Docker image with version
+on:
+  pull_request:
+    branches: [ master ]
+  push:
+    branches: [ master ]
+    tags:
+      - '*.*.*'
```

I would keep array formats consistent with either
```suggestion
  pull_request:
    branches: [ master ]
  push:
    branches: [ master ]
    tags: ['*.*.*']
```
or
```suggestion
  pull_request:
    branches:
    - master
  push:
    branches:
    - master
    tags:
    - '*.*.*'
```

---

also, does this  work with our release tags having the format `vx.x.x`?